src/main/kotlin/no/nb/mlt/wls/domain/WLSService.kt

```diff
@@ -93,27 +93,38 @@ class WLSService(
 
         itemIds.map { itemId ->
             val item = getItem(itemId.hostName, itemId.hostId)!!
-            if (itemsPickedMap[item.hostId] == null) {
-                // TODO - Review this. This state should generally not happen with the preconditions. Does it need to be handled at all?
-                throw ItemNotFoundException("Item with ID ${itemId.hostId} for host ${item.hostName} was not found, despite existing!")
-            }
 
-            val itemsStocked = item.quantity ?: 0.0
-            val itemsPicked = itemsPickedMap[item.hostId] ?: 0.0
+            val itemsInStockQuantity = item.quantity ?: 0.0
+            val pickedItemsQuantity = itemsPickedMap[item.hostId] ?: 0.0
 
             // In the case of over-picking, set quantity to zero
             // so that on return the database hopefully recovers
-            if (itemsPicked > itemsStocked) {
+            if (pickedItemsQuantity > itemsInStockQuantity) {
                 logger.error {
-                    "Tried to pick too many items for ${item.hostId}. WLS DB has $itemsStocked stocked, and storage system tried to pick $itemsPicked"
+                    "Tried to pick too many items for ${item.hostId}. " +
+                        "WLS DB has $itemsInStockQuantity stocked, and storage system tried to pick $pickedItemsQuantity"
                 }
             }
+            val quantity = Math.clamp(itemsInStockQuantity.minus(pickedItemsQuantity), 0.0, Double.MAX_VALUE)
+            val location: String =
+                if (quantity == 0.0) {
```

Comment:
Not fully sure if this will ever happen, but since floating point arithmetics can be a bit fucky we might end up with quantity being something like 0.0000000001 which won't equal 0.0 according to this check. I am unsure if either the `minus` or `clamp` functions handle that in any way to prevent this scenario.
If we want to be super sure we can add a simple extension function to Double class like:
```kotlin
infix fun Double.eq(other: Double) = abs(this - other) < 0.000001
```
And then use it here as `quantity eq 0.0`

No need to do it right now, but it might be something to think about

Reply:
We can also use BigDecimal which should give us accurate results